lib/elixir/lib/calendar.ex

```diff
@@ -95,10 +118,80 @@ defmodule Calendar do
   @callback naive_datetime_to_string(year, month, day, hour, minute, second, microsecond) :: String.t
 
   @doc """
-  Coverts the datetime (with time zone) into a string according to the calendar.
+  Converts the datetime (with time zone) into a string according to the calendar.
   """
   @callback datetime_to_string(year, month, day, hour, minute, second, microsecond,
                                time_zone, zone_abbr, utc_offset, std_offset) :: String.t
+
+  @doc """
+  Converts the time into a string according to the calendar.
+  """
+  @callback time_to_string(hour, minute, second, microsecond) :: String.t
+
+  @doc """
+  Converts the given datetime (with time zone)
+  into the internal Calendar.rata_die format.
+  """
+  @callback naive_datetime_to_rata_die(year, month, day, hour, minute, second, microsecond) :: rata_die
+
+  @doc """
+  Converts a datetime in the internal Calendar.rata_die format
+  to the Calendar's datetime (with time zone) format.
+  """
+  @callback naive_datetime_from_rata_die(rata_die) :: {year, month, day, hour, minute, second, microsecond}
+
+  @doc """
+  Converts the given time to the internal Calendar.day_fraction format.
+  """
+  @callback time_to_day_fraction(hour, minute, second, microsecond) :: day_fraction
+
+  @doc """
+  Converts the given time in Calendar.day_fraction format
+  to the Calendar's time format.
+  """
+  @callback time_from_day_fraction(day_fraction) :: {hour, minute, second, microsecond}
+
+  @doc """
+  The result of this function is used to check if two calendars roll over at the same time of day.
+  If they do not, we can only convert DateTimes and Times between them.
+  If they do, this means that we can also convert Dates and NaiveDateTimes between them.
+
+  This day fraction should be in its most simplified form possible, to make comparisons fast.
+
+  ## Some examples:
+
+  - If, in your Calendar, a new day starts at midnight, return {0, 1}.
```

Comment:
Please indent with two spaces each bullet and use `*` (which is the standard in Elixir source).